convert/convert.go

```diff
@@ -0,0 +1,180 @@
+package convert
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Clever/mongo-op-throttler/operation"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// OplogBytesToOp converts the raw bytes for an oplog into a Mongo operation
+// as defined by operation.Op. There are two reasons we don't immediately write
+// the oplog entries to the database.
+// 1. Keeps the logic for understanding oplogs separate from the rest of the code
+// 2. Makes it easier to have a worker that takes in a file of operation.Ops instead
+// of the oplog
+func OplogBytesToOp(raw []byte) (*operation.Op, error) {
+	var bsonOp bson.M
+	if err := bson.Unmarshal(raw, &bsonOp); err != nil {
+		return nil, fmt.Errorf("Error parsing bson: %s", err.Error())
+	}
+
+	return oplogEntryToOp(bsonOp)
+}
+
+// oplogEntryToOp converts from bson.M to operation.Op
+// Note that this has only been tested for the Mongo 2.4 format
+// Based on the logic from the source code:
+// https://github.com/mongodb/mongo/blob/v2.4/src/mongo/db/oplog.cpp#L791
+//
+// oplog entries have the following format:
+// "ts" : The timestampe of the entry
+// "h" :
```

Comment:
looks like `"h"` corresponds to the hash: https://github.com/mongodb/mongo/blob/v2.4/src/mongo/db/oplog.cpp#L201